redbot/core/utils/_internal_utils.py

```diff
@@ -240,7 +241,52 @@ async def create_backup(dest: Path = Path.home()) -> Optional[Path]:
 # this might be worth moving to `bot.send_to_owners` at later date
 
 
-async def send_to_owners_with_preprocessor(
+def send_to_owners_with_preprocessor(
+    bot: Red,
+    content: str,
+    *,
+    show_origin_message: bool = True,
+    content_preprocessor: Optional[
+        Callable[[Red, discord.abc.Messageable, str], Awaitable[str]]
+    ] = None,
+    **kwargs,
+):
+    """
+    This sends something to all owners and their configured extra destinations.
+
+    This acts the same as `Red.send_to_owners`, with
+    one added keyword argument as detailed below in *Other Parameters*.
+
+    Other Parameters
+    ----------------
+    content_preprocessor: Optional[Callable[[Red, discord.abc.Messageable, str], Awaitable[str]]]
+        Optional async function that takes
+        bot object, owner notification destination and message content
+        and returns the content that should be sent to given location.
+    """
+    try:
+        stack = inspect.stack()
+        if stack[1].function == "send_to_owners_with_prefix_replaced":
```

Comment:
Big performance boost, see the other review comment for more info.
We could probably gain even more, if we just had a kwarg in which we would pass the function name, since this utility function is not public anyway.
```suggestion
        frame = inspect.currentframe().f_back
        if frame.f_code.co_name == "send_to_owners_with_prefix_replaced":
```